RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCore30Platform.cs

```diff
@@ -0,0 +1,446 @@
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct CORINFO_SIG_INFO {
+            public int callConv; // CorInfoCallConv
+            public IntPtr retTypeClass; // CORINFO_CLASS_HANDLE // if the return type is a value class, this is its handle (enums are normalized)
+            public IntPtr retTypeSigClass; // CORINFO_CLASS_HANDLE // returns the value class as it is in the sig (enums are not converted to primitives)
+            public byte retType; // CorInfoType : 8
+            public byte flags; // unsigned : 8 // used by IL stubs code
+            public ushort numArgs; // unsigned : 16 
+            public CORINFO_SIG_INST sigInst; // information about how type variables are being instantiated in generic code
+            public IntPtr args; // CORINFO_ARG_LIST_HANDLE
+            public IntPtr pSig; // COR_SIGNATURE*
+            public uint sbSig;
+            public IntPtr scope; // CORINFO_MODULE_HANDLE // passed to getArgClass
+            public uint token; // mdToken (aka ULONG32 aka unsigned int)
+        }
+
+        [StructLayout(LayoutKind.Sequential)]
+        private unsafe struct CORINFO_METHOD_INFO {
+            public IntPtr ftn;   // CORINFO_METHOD_HANDLE
+            public IntPtr scope; // CORINFO_MODULE_HANDLE
+            public byte* ILCode;
+            public uint ILCodeSize;
+            public uint maxStack;
+            public uint EHcount;
+            public int options; // CorInfoOptions
+            public int regionKind; // CorInfoRegionKind
+            public CORINFO_SIG_INFO args;
+            public CORINFO_SIG_INFO locals;
+        }
+
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private unsafe delegate CorJitResult d_compileMethod(
+            IntPtr thisPtr, // ICorJitCompiler*
+            IntPtr corJitInfo, // ICorJitInfo*
+            in CORINFO_METHOD_INFO methodInfo, // CORINFO_METHOD_INFO*
+            uint flags,
+            out byte* nativeEntry,
+            out ulong nativeSizeOfCode
+        );
+
+        [ThreadStatic]
+        private static int hookEntrancy = 0;
+        private unsafe CorJitResult CompileMethodHook(
+            IntPtr jit, // ICorJitCompiler*
+            IntPtr corJitInfo, // ICorJitInfo*
+            in CORINFO_METHOD_INFO methodInfo, // CORINFO_METHOD_INFO*
+            uint flags, 
+            out byte* nativeEntry, 
+            out ulong nativeSizeOfCode) {
+
+            nativeEntry = null;
+            nativeSizeOfCode = 0;
+
+            if (jit == IntPtr.Zero)
+                return CorJitResult.CORJIT_OK;
+
+            hookEntrancy++;
+            try {
+                if (hookEntrancy == 1) {
+                    // This is the top level JIT entry point, do our custom stuff
+                    CorJitResult result = real_compileMethod.Invoke(jit, corJitInfo, methodInfo, flags, out nativeEntry, out nativeSizeOfCode);
```

Comment:
I think I did that just so that it was extra explicit that that was a delegate invocation, since its in a code block that we have to be extremely careful about what we do in.